packages/shared/scope-hoisting/src/formats/commonjs.js

```diff
@@ -276,13 +284,16 @@ export function generateExports(
         }
       }
     } else {
-      for (let {exportSymbol, symbol} of bundleGraph.getExportedSymbols(
+      for (let {exportSymbol, symbol, asset} of bundleGraph.getExportedSymbols(
         entry,
       )) {
-        if (symbol) {
-          symbol = replacements.get(symbol) || symbol;
+        if (!symbol) {
+          let relativePath = relative(options.inputFS.cwd(), asset.filePath);
+          throw new Error(`${relativePath} does not export '${exportSymbol}'`);
```

Comment:
Do we have enough information to show a diagnostic?

Reply:
Ideally we would include a codeFrame of the relevant sections, but that isn't possible without your AST reusing PR.

I've now added a diagnostic with `messages` and `filePath` and modified `prettyDiagnostic` to use `filePath` as a fallback for `stack` (@DeMoorJasper ?):

![Bildschirmfoto 2020-02-05 um 16 14 23](https://user-images.githubusercontent.com/4586894/73854320-a3afe600-4832-11ea-9340-7d2130ca0220.png)